rosidl_typesupport_opensplice_cpp/resource/msg__type_support.cpp.template

```diff
@@ -380,7 +380,9 @@ static message_type_support_callbacks_t callbacks = {
   "@(spec.base_type.type)",
   &register_type__@(spec.base_type.type),
   &publish__@(spec.base_type.type),
-  &take__@(spec.base_type.type)
+  &take__@(spec.base_type.type),
+  nullptr,  // convert ros to dds (handled differently for C++)
```

Comment:
This is part of the work to reuse publish and take between the two. I'd rather not create more similar but different structures, but I agree that it is a bit strange at the moment.